File: solutions/javascript/hamming/1/hamming.js
export function compute(strand1, strand2) {
    const length = strand1.length;
    let differentPieces = 0;
    if (length !== strand2.length)
        throw (new Error('strands must be of equal length'));
    for (let index = 0; index < length; index++)
        strand1[index] !== strand2[index] ? differentPieces++ : null;
    return differentPieces;
};

File: solutions/lua/hamming/1/hamming.lua
local Hamming = {}

function Hamming.compute(a, b)
    distance = 0
    if #a ~= #b then return -1 end
    for index = 1, #a do
        if a:sub(index, index) ~= b:sub(index, index) then distance = distance + 1 end
    end
    return distance
end

return Hamming